packages/web-config-server/src/apiV1/dataBuilders/helpers/calculateOperationForAnalytics.js

```diff
@@ -0,0 +1,94 @@
+import { checkValueSatisfiesCondition, replaceValues } from '@tupaia/utils';
+import { NO_DATA_AVAILABLE } from '/apiV1/dataBuilders/constants';
+import { divideValues } from './divideValues';
+import { subtractValues } from './subtractValues';
+
+const checkCondition = (value, config) =>
+  checkValueSatisfiesCondition(value, config.condition) ? 'Yes' : 'No';
+
+const formatString = (value, config) => replaceValues(config.format, { value });
+
+const performSingleAnalyticOperation = (analytics, config) => {
+  const { operator, dataElement } = config;
+  const filteredAnalytics = analytics.filter(({ dataElement: de }) => de === dataElement);
+  if (filteredAnalytics.length > 1) {
+    throw new Error(`Too many results passed to checkConditions (calculateOperationForAnalytics)`);
+  } else if (filteredAnalytics.length === 0) {
+    return NO_DATA_AVAILABLE;
+  }
+  return OPERATORS[operator](filteredAnalytics[0].value, config);
+};
+
+const sumDataValues = (analytics, dataValues) => {
+  let sum; //Keep sum undefined so that if there's no data values then we can distinguish between No data and 0
+
+  analytics.forEach(({ dataElement, value }) => {
+    if (dataValues.includes(dataElement)) {
+      sum = (sum || 0) + (value || 0);
+    }
+  });
+
+  return sum;
+};
+
+const performArithmeticOperation = (analytics, arithmeticConfig) => {
+  const { operator, operands: operandConfigs } = arithmeticConfig;
+
+  if (!operandConfigs || operandConfigs.length < 2) {
+    throw new Error(`Must have 2 or more operands`);
+  }
+
+  const operands = operandConfigs.map(operandConfig =>
+    sumDataValues(analytics, operandConfig.dataValues),
+  );
+
+  let result = operands[0];
+  const operationMethod = OPERATORS[operator];
+
+  for (let i = 1; i < operands.length; i++) {
+    const currentOperand = operands[i];
+    result = operationMethod(result, currentOperand);
+  }
+
+  return result;
+};
+
+const combineBinaryIndicatorsToString = (analytics, config) => {
+  const { dataElementToString } = config;
+  const filteredAnalytics = analytics.filter(({ dataElement: de }) =>
+    Object.keys(dataElementToString).includes(de),
+  );
+  const stringArray = [];
+  filteredAnalytics.forEach(({ dataElement, value }) => {
+    const stringValue = value === 'Yes' ? dataElementToString[dataElement] : '';
```

Comment:
I think it would be super helpful if we can have this configurable instead of always checking for 'Yes'. But it shouldn't be too hard to extend when someone needs it so don't think it's worth changing now

Reply:
Good point! But I agree. Also ideal would be using this logic in the existing measureBuilder, or even being able to remove that measureBuilder entirely.